core/src/main/java/org/apache/iceberg/SnapshotProducer.java

```diff
@@ -116,6 +117,38 @@ public ThisT scanManifestsWith(ExecutorService executorService) {
     return self();
   }
 
+  @Override
+  public ThisT toBranch(String branch) {
+    throw new UnsupportedOperationException("Performing operations on a branch is currently not supported");
```

Comment:
Minor: I think this is a good time to use `this.getClass().getName()` so that it's clear which operation doesn't support committing to a branch.

```java
throw new UnsupportedOperationException(Sting.format(
    "Cannot commit to branch %s: %s does not support branch commits", branch,
    this.getClass().getName());
```